src/main/kotlin/com/sourcegraph/cody/autocomplete/CodyAutocompleteManager.kt

```diff
@@ -158,7 +158,8 @@ class CodyAutocompleteManager {
     val originalText = editor.document.getText(TextRange(offset - caretPositionInLine, offset))
 
     val prefixStartPosition = maxOf(originalText.lastIndexOf("."), originalText.lastIndexOf(" "), 0)
-    if (!lookupString.isNullOrEmpty() &&
+    if (prefixStartPosition < originalText.length &&
```

Comment:
Based on the original stack trace, this happens with empty strings

```
@ "".subSequence(1, 0)
java.lang.StringIndexOutOfBoundsException: begin 1, end 0, length 0
```


```suggestion
    if (prefixStartPosition + 1 < originalText.length &&
```

Reply:
I was crashing on this line `originalText.subSequence(prefixStartPosition + 1, originalText.length))`
when `originalText.length = 0` and `prefixStartPosition + 1 = 1` (so `prefixStartPosition = 0`).
Thus I think condition I added is valid and sufficient to prevent crash.
That said it can still result in producing empty subSequence and in such case `startsWith` will be always try no matter what `lookupString` is.
I would say let's double check and merge on Monday.

Reply:
Agree the current diff fixes the case when `originalText.length === 0`. Still, the logic reads like the bug below
```java
if (index < text.length) {
  text.charAt(index + 1)
}
```
My brain is just wired to not allow that kind of code.

Either way, a much cleaner fix is to use the `startsWith()` overload from the Kotlin stdlib that accepts a `startIndex` parameter, and remove `.subSequence()`. The original code was written in Java where that overload doesn't exist

Reply:
That would be great idea, but `Returns true if a substring of this string starting at the specified offset [startIndex](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/starts-with.html#kotlin.text$startsWith(kotlin.String,%20kotlin.String,%20kotlin.Int,%20kotlin.Boolean)/startIndex) starts with the specified prefix.`
So `startIndex` would affect position in `lookupString` while we want it to affect a prefix (`originalText`).
Truth to be told I see at least one more bug there and I do not fully understand what that code is supposed to do exactly, so maybe let's discuss offline.